app/2.0/docs/devguide/dom-template.md

```diff
@@ -83,4 +155,138 @@ use the standard DOM `querySelector`  method.
 
 
 
+## Remove empty text nodes {#strip-whitespace}
+
+
+Add the `strip-whitespace` boolean attribute to a template to remove
+any empty text nodes from the template's contents. This can result in a
+minor performance improvement.
+
+With empty text nodes:
+
+
+```html
+<dom-module id="has-whitespace">
+  <template> <div>A</div> <div>B</div> </template>
+  <script>
+    class HasWhitespace extends Polymer.Element {
+      static get is() { return  'has-whitespace' }
+      ready() {
+        super.ready();
+        console.log(this.shadowRoot.childNodes.length); // 5
+      }
+    }
+    customElements.define(HasWhitespace.is, HasWhitespace);
+  </script>
+</dom-module>
+```
+
+
+Without empty text nodes:
+
+
+```html
+<dom-module id="no-whitespace">
+  <template strip-whitespace>
+    <div>A</div>
+    <div>B</div>
+  </template>
+  <script>
+    class NoWhitespace extends Polymer.Element {
+      static get is() { return  'no-whitespace' }
+      ready() {
+        super.ready();
+        console.log(this.shadowRoot.childNodes.length); // 2
+      }
+    }
+    customElements.define(NoWhitespace.is, NoWhitespace);
+  </script>
+</dom-module>
+```
+
+## Preserve template contents
+
+Polymer performs one-time processing on any Polymer-managed templates, including templates nested
```

Comment:
That is, `preserve-content` does more than just avoiding the `.content` being cached & removed, it makes Polymer totally skip processing on it, so it won't get bindings, etc.